spec/immutable-data-structures.html

```diff
@@ -27,6 +39,32 @@ <h1>Properties of the Record Constructor</h1>
         <li>has a [[Prototype]] internal slot whose value is %Record.prototype%.</li>
         <li>has the following properties:</li>
       </ul>
+      <emu-clause id="sec-record.fromentries">
+        <h1>Record.fromEntries ( _iterable_ )</h1>
+        <p>The *fromEntries* function takes one argument _iterable_, and performs the following steps:</p>
+        <emu-alg>
+          1. Perform ? RequireObjectCoercible(_iterable_).
+          1. Let _fields_ be a new empty List.
+          1. Let _iteratorRecord_ be ? GetIterator(_iterable_).
+          1. Repeat,
+            1. Let _next_ be ? IteratorStep(_iteratorRecord_).
+            1. If _next_ is *false*,
+              1. Return a new Record whose [[Fields]] is _fields_.
+            1. Let _nextItem_ be ? IteratorValue(_next_).
+            1. If Type(_nextItem_) is not Object, then
+              1. Let _error_ be ThrowCompletion(a newly created *TypeError* object).
+              1. Return ? IteratorClose(_iteratorRecord_, _error_).
+            1. Let _k_ be Get(_nextItem_, *"0"*).
+            1. If _k_ is an abrupt completion, return ? IteratorClose(_iteratorRecord_, _k_).
+            1. Let _v_ be Get(_nextItem_, *"1"*).
+            1. If _v_ is an abrupt completion, return ? IteratorClose(_iteratorRecord_, _v_).
+            1. Let _field_ be { [[Key]]: _k_, [[Value]]: _v_ }.
+            1. Append _field_ to the end of list _fields_.

```

Comment:
This seems like it could benefit from using an abstract operation like AddEntriesFromIterable, where the `adder` creates a List, and then at the end, return the new Record?

Reply:
I based this originally on `AddEntriesFromIterable` in the first place, so that is fine by me. I initially simplified this to remove an additional abstract op since the resulting op would initially be only used here. I guess something like `CreateRecordFieldsFromIterable` (as you said, returning a list of fields)?

Reply:
The main difference is the "checking if it's an object and throwing". However if the `adder` did this check before appending to a list, it seems like it would work?